src/components/PathwayExercise/ContentEdit/editor.js

```diff
@@ -0,0 +1,276 @@
+  const ReactEditorJS = createReactEditorJS();
+  const handleInitialize = async (instance) => {
+    await setEditor(instance);
+    console.log(editor, "editor");
+  };
+
+  console.log("course", course);
+  console.log("id in editor", id);
+  let style = "";
+  let items = [];
+
+  let courseData = course.map((item, index) => {
+    let youtube;
+    if (item.component === "youtube") {
+      if (!item.value.includes("=")) {
+        youtube = "https://www.youtube.com/embed/" + item.value;
+      } else {
+        let value = item.value.split("=")[1];
+        youtube = "https://www.youtube.com/embed/" + value;
+      }
+    } else if (item.component === "text") {
+      if ("decoration" in item) {
+        if (item.decoration.type == "bullet") {
+          style = "unordered";
+        } else {
+          style = "ordered";
+        }
+        items.push(item.value);
+      } else {
+        items = [];
+      }
+    } else {
+      items = [];
+    }
+
+    let type;
+    let code;
+    let content = [];
+    if (item.component == "text" && "decoration" in item) {
+      type = "list";
+    } else if (item.component == "text") {
+      type = "paragraph";
+    } else if (item.component === "youtube") {
+      type = "embed";
+    } else if (item.component === "code") {
+      code = item.value.replace(/<br>/g, "\n").replace(/&emsp;/g, "    ");
+      type = item.component;
+    } else if (item.component === "table") {
+      type = "table";
+      const header = item.value.map((item) => item.header);
+      const allData = item.value.map((item) => item.items);
+      const dataInCol = allData[0].map((_, i) =>
+        allData.map((_, j) => allData[j][i])
+      );
+      content = [[...header], ...dataInCol];
+      console.log("allData", allData);
+      console.log("header", header);
+      console.log("dataInCol", dataInCol);
+      console.log("content", content);
+    } else {
+      type = item.component;
+    }
+
+    return {
+      type: type,
+      data: {
+        style: style,
+        items: items,
+        text:
+          type == "list"
+            ? false
+            : (item.component === "text" || item.component === "header") &&
+              item.value,
+        level: item.variant,
+        code: item.component === "code" && code,
+        file: {
+          url: item.component === "image" && item.value,
+        },
+        caption: item.component === "image" && item.alt,
+        withBorder: false,
+        withBackground: false,
+        stretched: true,
+        embed: item.component === "youtube" && youtube,
+        source: item.component === "youtube" && youtube,
+        service: item.component === "youtube" && item.value && item.component,
+        height: 320,
+        width: 580,
+        // withHeadings: true,
+        // content: content,
+      },
+    };
+  });
+
+  let blocks = courseData.filter((item, index) => {
+    const stringifiedItem = JSON.stringify(item);
```

Comment:
Using `JSON.stringify` repeatedly for large objects can negatively affect performance and is unnecessary in this case since we only need to compare the `items` properties of the objects. Here, we can use `===` for that as `items` will refer to the same objects. Additionally, note that while not relevant here, there could be unexpected results with this approach so when comparing objects for objects, it's better to use built-in tested functions such as Lodash's `isEqual` (https://lodash.com/docs/4.17.15#isEqual):
```
JSON.stringify({f: x => 2*x}) === JSON.stringify({}) // true, function isn't serialized
JSON.stringify({a: 42, b: 0}) === JSON.stringify({b: 0, a: 42}) // false, order of keys is different
```

)

Reply:
I had tried with `isEqual` Lodash but got confused that, which 2 objects I needed to compare.

Reply:
It should be the same ones for which you're comparing the values of `JSON.stringify`, so below you could use `_.isEqual(obj, item)` instead.